iris-mpc-cpu/src/genesis/hawk_handle.rs

```diff
@@ -167,7 +167,24 @@ impl Handle {
         let results_ = parallelize(jobs_per_side.into_iter()).await?;
         let results: [_; 2] = results_.try_into().unwrap();
 
-        Ok(JobResult::new(request, HawkMutation(results)))
+        // Convert the results into SingleHawkMutation format
+        let [left_plans, right_plans] = results;
+        assert_eq!(left_plans.len(), right_plans.len());
+        let mut mutations = Vec::new();
+
+        for i in 0..left_plans.len() {
+            let left_plan = left_plans.get(i).cloned().flatten();
```

Comment:
To clarify, it never makes sense to have only one side. It would be clearer to remove the `Option` as soon as possible rather than propagating them into `SingleHawkMutation`.